### ISSUE TITLE
TypeAliasType does not reject some illegal expression forms

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6836]
The following should generate a type error but does not:

```python
BadAlias19 = TypeAliasType("BadAlias19", list or set)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.344, which I just published. It will also be included in a future release of pylance.
